packages/ui/src/components/Modal/useModal.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useCallback, useEffect, useImperativeHandle, useRef, useState } from 'react'
+
+import { Modal } from './modal'
+
+import { usePatchElement } from '../_utils/hooks/usePatchElement'
+
+import type { ModalProps } from './modal'
+
+type AnyArgs = Array<any>
```

Comment:
do we really have no idea what args will be?

Reply:
Antd does not, that's part of their original code

Reply:
and I guess they're taking logic from `rc-dialog`, so it's nested type def, that's why it's any